extension-style/src/main/java/com/mapbox/maps/extension/style/expressions/generated/Expression.kt

```diff
@@ -921,6 +921,55 @@ class Expression : Value {
       this@ExpressionBuilder.apply(block)
     }
 
+    /**
+     * Add a pair of input and output values.
+     * The `input` may be any numeric expression (e.g., `["get", "population"]`).
+     * Stop inputs must be numeric literals in strictly ascending order.
+     * Returns the output value of the stop just less than the input,
+     * or the first output if the input is less than the first stop.
+     *
+     * @param key input as a string
+     * @param expression as Expression
+     * It is to be used as part of parameters in the step expression.
+     */
+    fun stop(key: String, expression: Expression) {
+      stop(Companion.literal(key), expression)
+    }
+
+    /**
+     * Add a pair of input and output values.
+     * The `input` may be any numeric expression (e.g., `["get", "population"]`).
+     * Stop inputs must be numeric literals in strictly ascending order.
+     * Returns the output value of the stop just less than the input,
+     * or the first output if the input is less than the first stop.
+     *
+     * @param key input as a double
+     * @param expression as Expression
+     * It is to be used as part of parameters in the step expression.
+     */
+    fun stop(key: Double, expression: Expression) {
+      stop(Companion.literal(key), expression)
+    }
+
+    /**
+     * Add a pair of input and output values.
+     * The `input` may be any numeric expression (e.g., `["get", "population"]`).
+     * Stop inputs must be numeric literals in strictly ascending order.
+     * Returns the output value of the stop just less than the input,
+     * or the first output if the input is less than the first stop.
+     *
+     * @param expressions to be applied for stop
+     * @return Expression
+     * It is to be used as part of parameters in the step expression.
+     */
+    fun stop(vararg expressions: Expression): Expression {
```

Comment:
here, the user can provide expressions as input and output for stop expression in order to simplify. Although I am not sure if this can lead to other vulnerabilities, in case the expressions are not the expected expressions. do you think we should keep it as it is or make it private and just expose the strongly typed overloads, wdyt @mapbox/maps-android

Reply:
I don't think we should expose this function, each stop should only contain two parameters.